packages/firestore/src/api/reference_impl.ts

```diff
@@ -287,10 +287,26 @@ export function getDocsFromServer<
  * Writes to the document referred to by this `DocumentReference`. If the
  * document does not yet exist, it will be created.
  *
+ * Note that the returned `Promise` does _not_ resolve until the data is
+ * successfully written to the remote Firestore backend and, similarly, is not
+ * rejected until the remote Firestore backend reports an error saving the given
+ * data. So if the client cannot reach the backend, for example due to being
+ * offline, then the returned `Promise` will not resolve for a potentially-long
+ * time, for example until the client has gone back online. That being said,
+ * the given data _will_ be immediately saved to the local cache and will be
+ * incorporated into future "get" operations as if it had been successfully
+ * written to the remote Firestore server. The data will _eventually_ be written
```

Comment:
You could reference this behavior as latency compensation. We do [use this term in public documentation](https://firebase.google.com/docs/firestore/query-data/listen#events-local-changes), so giving it a name here may make it easier to search for.